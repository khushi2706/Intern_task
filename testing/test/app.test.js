const request = require("supertest")
const koa = require("koa")
const axios = require("axios")

//const route = require("../server")
describe("get Endpoints", () => {
  let server
  it("get hi", async () => {
    const res = await axios.get("http://localhost:5000/")
    expect(res.status).toEqual(200)
    expect(res.data).toEqual({ msg: "Hi" })
    expect(res.data).toMatchSnapshot()
  })
  it("post data", async () => {
    const data = {
      name: "khushi",
      age: 20,
    }
    const res = await axios.post("http://localhost:5000/data", data)
    expect(res.status).toEqual(200)
    expect(res.data).toEqual({ success: true })
    //  expect(res.data).toMatchSnapshot()
  })
  it("should sucess false", async () => {
    const data = {
      name: "khushi",
    }
    const res = await axios
      .post("http://localhost:5000/data", data)
      .catch((error) => {
        expect(error.response.status).toEqual(400)
        expect(error.response.data.success).toEqual(false)
      })
  })
})
