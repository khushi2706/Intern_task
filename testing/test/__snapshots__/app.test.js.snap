// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get Endpoints get hi 1`] = `
{
  "msg": "Hi",
}
`;

exports[`get Endpoints post data 1`] = `
{
  "success": true,
}
`;
